openstack/containerorchestration/v1/common/errors.go

```diff
@@ -0,0 +1,118 @@
+package common
+
+import (
+	"encoding/json"
+	"github.com/gophercloud/gophercloud"
+	"strings"
+)
+
+// ErrorResponse represents failed response
+type ErrorResponse struct {
+	gophercloud.ErrUnexpectedResponseCode
+	Errors []Error `json:"errors" required:"true"`
+}
+
+// Error represents a Magnum API error message
+type Error struct {
+	Status    int    `json:"status"`
+	Code      string `json:"code"`
+	Title     string `json:"title"`
+	Detail    string `json:"detail"`
+	RequestID string `json:"request_id,omitempty"`
+}
+
+// Error returns the error message details from the server
+func (e *ErrorResponse) Error() string {
+	errors := []string{}
+	for _, e := range e.Errors {
+		errors = append(errors, e.Detail)
+	}
+	return strings.Join(errors, "\n")
+}
+
+func (e *ErrorResponse) unwrapError(r gophercloud.ErrUnexpectedResponseCode) bool {
```

Comment:
> That would need to be for every result type because Magnum returns that same error structure for all failed operations (create/delete/update/get).

Ah, OK. That would be tedious.

> I'll update the PR with your suggestion but for every result type, not just Delete so you can look it over.

No, that's OK. Instead, what about having that `ExtractErr` method on something like `ErrResult` and having all the results embed that type?

```go
type DeleteResult struct {
  ErrResult
}

type ErrResult struct {
  gophercloud.ErrResult
}

func (h *ErrResult) ExtractErr() error {
  if h.Err != nil {
    if e, ok := unwrapError(h.Err); ok {
      return e
    }
    return h.Err
  }
  return nil
}
```